src/adapter.ts

```diff
@@ -52,8 +83,8 @@ class MongooseAdapter {
    * Opens a connection to mongoDB
    * @returns {Promise<void>}
    */
-  async _open () {
-    await mongoose.connect(this.uri, this.options)
+  async _open() {
+    await connect(this.uri, this.options)
```

Comment:
This could be changed to `this.instance = await connect(this.uri, this.options)`, now that I think about it.

Reply:
@Zxilly

Reply:
I agree if moongoose could automatic reconnect. I will working on that.

Reply:
Automatic reconnection is done by mongodb client by default (which mongoose uses). It tries 30 times by default and value can be increased using different config value on initial connect.

Since this function is primarily used for initial connection when the Adapter is created, I think we can remove this function completely.

Reply:
@Sefriol OK, since this PR has been merged. I will later create another PR to fix these problems